cluster/store/service.go

```diff
@@ -298,40 +299,40 @@ func (s *Service) WaitUntilDBRestored(ctx context.Context, period time.Duration)
 
 // QueryReadOnlyClass will verify that class is non empty and then build a Query that will be directed to the leader to
 // ensure we will read the class with strong consistency
-func (s *Service) QueryReadOnlyClass(class string) (*models.Class, uint64, error) {
-	if class == "" {
-		return &models.Class{}, 0, fmt.Errorf("empty class name: %w", errBadRequest)
+func (s *Service) QueryReadOnlyClasses(classes ...string) (map[string]classcache.VersionedClass, error) {
+	if len(classes) == 0 {
+		return nil, fmt.Errorf("empty class name: %w", errBadRequest)
 	}
 
 	// Build the query and execute it
-	req := cmd.QueryReadOnlyClassRequest{Class: class}
+	req := cmd.QueryReadOnlyClassesRequest{Classes: classes}
 	subCommand, err := json.Marshal(&req)
 	if err != nil {
-		return &models.Class{}, 0, fmt.Errorf("marshal request: %w", err)
+		return nil, fmt.Errorf("marshal request: %w", err)
 	}
 	command := &cmd.QueryRequest{
-		Type:       cmd.QueryRequest_TYPE_GET_CLASS,
+		Type:       cmd.QueryRequest_TYPE_GET_CLASSES,
 		SubCommand: subCommand,
 	}
 	queryResp, err := s.Query(context.Background(), command)
 	if err != nil {
-		return &models.Class{}, 0, fmt.Errorf("failed to execute query: %w", err)
+		return nil, fmt.Errorf("failed to execute query: %w", err)
```

Comment:
You always want to return `&models.Class{}` here and non-nil as otherwise it might trigger marshalling error further down the line. IIRC that was the reason why all these were initialized. If your change doesn't trigger any error (should trigger with `store_test.go` then feel free to ignore that comment :)

Reply:
i think it's not anymore because it returns a map now